plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java

```diff
@@ -683,4 +684,24 @@ private BaseTable loadTable(String tableName)
     {
         return IcebergTestUtils.loadTable(tableName, metastore, fileSystemFactory, "hive", "tpch");
     }
+
+    @Test
+    public void testPartitionColumns()
+    {
+        String tableName = "test_partition_columns_" + randomNameSuffix();
+        assertUpdate(String.format("""
+                CREATE TABLE %s WITH (partitioning = ARRAY[
+                    '"r1.f1"',
+                    'bucket(b1, 4)'
+                ]) AS
+                SELECT
+                    CAST('c1' AS VARCHAR) as c1
+                    , CAST(ROW(1, 2) AS ROW(f1 integer, f2 integer)) as r1
+                    , CAST('2022-01-01 01:01:01' AS TIMESTAMP) as d1
+                    , CAST('2022-01-01 01:01:01' AS TIMESTAMP) as d2
+                    , CAST('2022-01-01 01:01:01' AS TIMESTAMP) as d3
+                    , CAST('2022-01-01 01:01:01' AS TIMESTAMP) as d4
+                    , CAST('b' AS VARCHAR) as b1
+                    , CAST('12345678' AS VARCHAR) as t1""", tableName), 1);
```

Comment:
Please verify the result with `SELECT` statement.

Reply:
Added `select from $partitions`